### PR TITLE
[WTF-2074] Rollup 3 Upgrade

### PR DESCRIPTION
**Pull request by weirdwater**
## Checklist

-   Contains unit tests ✅ ❌
-   Contains breaking changes ✅ 
-   Compatible with: MX 🔟 
-   Did you update version and changelog? ✅ 
-   PR title properly formatted (`[XX-000]: description`)? ✅ 

## This PR contains

-   [ ] Bug fix
-   [ ] Feature
-   [x] Refactor
-   [x] Documentation
-   [ ] Other (describe)

## What is the purpose of this PR?

This PR aims to upgrade the bundler setup in order to offer more recent packages and extend the longevity of the package.

## Relevant changes

Package versions have been updated and modifications have been made to the Rollup and jest configurations to be compatible with newer versions.

## What should be covered while testing?

How to test the branch locally:

1. Pull repository and checkout this branch `wtf/rollup-3`
2. Run `npm pack`
3. Copy the full path of the resulting archive (The previous command states the name of the file)
4. In the widget directory, run `npm install $full_path_to_archive`

Properties to check:

- Newly generated widgets should build and test without any modification necessary. 
- Widgets without custom configuration should build and test without any modification necessary.
- Widgets with custom configuration (for example [carousel-web](https://github.com/mendix/web-widgets/blob/main/packages/pluggableWidgets/carousel-web/rollup.config.js)) should be upgrade-able using the guide in CHANGELOG.md

### CODE REVIEW

File: packages/pluggable-widgets-tools/configs/rollup.config.mjs [hunk 1]
```diff
@@ -146,6 +150,7 @@ export default async args => {
                 sourcemap: !production ? "inline" : false
             },
             external: commonExternalLibs,
+            strictDeprecations: true,
```
Comment (alihcsumer) [inline packages/pluggable-widgets-tools/configs/rollup.config.mjs:153]: to be removed.

File: packages/pluggable-widgets-tools/configs/rollup.config.mjs [hunk 2]
```diff
@@ -178,6 +183,7 @@ export default async args => {
                 sourcemap: false
             },
             external: commonExternalLibs,
+            strictDeprecations: true,
```
Comment (alihcsumer) [inline packages/pluggable-widgets-tools/configs/rollup.config.mjs:186]: as well

File: packages/pluggable-widgets-tools/configs/rollup.config.mjs (hunk 1 above)
Comment (weirdwater) [reply packages/pluggable-widgets-tools/configs/rollup.config.mjs]: Done

File: packages/pluggable-widgets-tools/configs/rollup.config.mjs (hunk 2 above)
Comment (weirdwater) [reply packages/pluggable-widgets-tools/configs/rollup.config.mjs:184]: And done

**Review by legio-vi-ferrata — APPROVED** [review]

**Review by alihcsumer — APPROVED** [review]

**Pull request merged**